Word Search.java

```diff
@@ -0,0 +1,81 @@
+/*
+class Solution:
+    def exist(self, board: List[List[str]], word: str) -> bool:
+        if board is None or len(board) == 0:
+            return False
+        
+        self.dirs = [[1,0], [0,1], [-1,0], [0,-1]]
+        rows, cols = len(board), len(board[0])
+        for i in range(rows):
+            for j in range(cols):
+                if self.backtrack(board, word, i, j, 0):
+                    return True
+        
+        return False
+    def backtrack(self, board, word, row, col, index):
+        if index == len(word):
+            return True
+        if row < 0 or row > len(board)-1 or col < 0 or col > len(board[0])-1:
+            return False
+        
+        if board[row][col] == word[index]:
+            temp = board[row][col]
+            board[row][col] = "#"
+            for r,c in self.dirs:
+                new_row = r + row
+                new_col = c + col
+                
+                if self.backtrack(board, word, new_row, new_col, index+1):
+                    board[row][col] = temp
+                    return True
+            
+            board[row][col] = temp
+        
+        return False
+*/
+
+// Time Complexity : O(4^l) where l is length of word
+// Space Complexity : O(l) 
```

Comment:
The time and space here should be O(n * m * 3L), Where n and m are the dimensions of the board respectively.